pkgs/by-name/ol/olympus/package.nix

```diff
@@ -0,0 +1,170 @@
+{
+  lib,
+  fetchFromGitHub,
+  fetchzip,
+  buildDotnetModule,
+  mono4,
+  love,
+  lua51Packages,
+  msbuild,
+  sqlite,
+  curl,
+  libarchive,
+  buildFHSEnv,
+  xdg-utils,
+}:
+# WONTFIX: On NixOS, cannot launch Steam installations of Everest / Celeste from Olympus.
+# The way it launches Celeste is by directly executing steamapps/common/Celeste/Celeste,
+# and it does not work on NixOS (even with steam-run).
+# This should be considered a bug of Steam on NixOS (and is probably very hard to fix).
+# https://github.com/EverestAPI/Olympus/issues/94 could be a temporary fix
+
+# FIXME: olympus checks if xdg-mime x-scheme-handler/everest for a popup. If it's not set it complains about it.
+# I'm pretty sure thats by user so end user needs to do it
+
+let
+  lua-subprocess = lua51Packages.buildLuarocksPackage {
+    pname = "subprocess";
+    version = "bfa8e9";
+    src = fetchFromGitHub {
+      owner = "0x0ade"; # a developer of Everest
+      repo = "lua-subprocess";
+      rev = "bfa8e97da774141f301cfd1106dca53a30a4de54";
+      hash = "sha256-4LiYWB3PAQ/s33Yj/gwC+Ef1vGe5FedWexeCBVSDIV0=";
+    };
+    rockspecFilename = "subprocess-scm-1.rockspec";
+  };
+
+  # NOTE: on installation olympus uses MiniInstallerLinux which is dynamically linked, this makes it run fine
+  fhs-env = buildFHSEnv {
+    name = "olympus-fhs";
+    targetPkgs =
+      pkgs:
+      (with pkgs; [
+        icu
+        stdenv.cc.cc
+        libgcc.lib
+        openssl
+      ]);
+    runScript = "bash";
+  };
+
+  lsqlite3 = lua51Packages.buildLuarocksPackage {
+    pname = "lsqlite3";
+    version = "0.9.6-1";
+    src = fetchzip {
+      url = "http://lua.sqlite.org/index.cgi/zip/lsqlite3_v096.zip";
+      hash = "sha256-Mq409A3X9/OS7IPI/KlULR6ZihqnYKk/mS/W/2yrGBg=";
+    };
+    buildInputs = [ sqlite.dev ];
+  };
+
+  dotnet-out = "sharp/bin/Release/net452";
+  pname = "olympus";
+  phome = "$out/lib/${pname}";
+  nfd = lua51Packages.nfd;
+in
+buildDotnetModule rec {
+  inherit pname;
+
+  # FIXME: I made up this version number.
+  version = "24.07.06.02";
```

Comment:
As per [nixpkgs guidelines](https://github.com/NixOS/nixpkgs/blob/8f28bd7f1bb16e51be6688ae1bdc0c9bfe4af717/pkgs/README.md#versioning), this should be `21.01.11.07-unstable-2024-07-06`

Reply:
> As per [nixpkgs guidelines](https://github.com/NixOS/nixpkgs/blob/8f28bd7f1bb16e51be6688ae1bdc0c9bfe4af717/pkgs/README.md#versioning), this should be `21.01.11.07-unstable-2024-07-06`

I believe Olympus does have official versioning, listed on the Azure pipeline page [here](https://dev.azure.com/EverestAPI/Olympus/_build?definitionId=4).

Reply:
I see, I saw the comment and didn't dig too far. There are no tags in the repo.

The gha page you linked appears to be a mirror of the azure pipeline, version numbers are defined here as `$(Date:yy.MM.dd)$(Rev:.rr)`:
https://github.com/EverestAPI/Olympus/blob/3ab5d063bb3eef815dbf6bb76e0d225af5f814be/azure-pipelines.yml#L56
The `$(Rev:.rr)` seems to mean "the nth daily run, padded to two digits" (https://learn.microsoft.com/en-us/azure/devops/pipelines/process/run-number?view=azure-devops).

The comment should probably be updated with information on how to source the correct version number.

Reply:
I as co-maintainer already wrote an update script that can figure out the version number automatically using Azure API and sent that to Petingoso. This will be fixed.